src/sys/termios.rs

```diff
@@ -125,6 +125,83 @@ mod ffi {
             pub c_ospeed: speed_t
         }
 
+        #[derive(Clone, Copy, Debug, PartialEq)]
+        pub enum BaudRate {
+            B0,
+            B50,
+            B75,
+            B110,
+            B134,
+            B150,
+            B200,
+            B300,
+            B600,
+            B1200,
+            B1800,
+            B2400,
+            B4800,
+            B9600,
+            B19200,
+            B38400,
+            B7200,
+            B14400,
+            B28800,
+            B57600,
+            B76800,
+            B115200,
+            B230400,
+            #[cfg(any(target_os = "netbsd", target_os = "freebsd"))]
+            B460800,
+            #[cfg(any(target_os = "netbsd", target_os = "freebsd"))]
+            B921600,
+        }
+
+        impl From<speed_t> for BaudRate {
+            fn from(s: speed_t) -> BaudRate {
+
+                use libc::{
+                    B0, B50, B75, B110, B134, B150,
+                    B200, B300, B600, B1200, B1800, B2400,
+                    B4800, B9600, B19200, B38400,
+                    B7200, B14400, B28800, B57600,
+                    B76800, B115200, B230400};
+
+                #[cfg(any(target_os = "netbsd", target_os = "freebsd"))]
+                use libc::{B460800, B921600};
+
+                match s {
+                    B0 => BaudRate::B0,
+                    B50 => BaudRate::B50,
+                    B75 => BaudRate::B75,
+                    B110 => BaudRate::B110,
+                    B134 => BaudRate::B134,
+                    B150 => BaudRate::B150,
+                    B200 => BaudRate::B200,
+                    B300 => BaudRate::B300,
+                    B600 => BaudRate::B600,
+                    B1200 => BaudRate::B1200,
+                    B1800 => BaudRate::B1800,
+                    B2400 => BaudRate::B2400,
+                    B4800 => BaudRate::B4800,
+                    B9600 => BaudRate::B9600,
+                    B19200 => BaudRate::B19200,
+                    B38400 => BaudRate::B38400,
+                    B7200 => BaudRate::B7200,
+                    B14400 => BaudRate::B14400,
+                    B28800 => BaudRate::B28800,
+                    B57600 => BaudRate::B57600,
+                    B76800 => BaudRate::B76800,
+                    B115200 => BaudRate::B115200,
+                    B230400 => BaudRate::B230400,
+                    #[cfg(any(target_os = "netbsd", target_os = "freebsd"))]
+                    B460800 => BaudRate::B460800,
+                    #[cfg(any(target_os = "netbsd", target_os = "freebsd"))]
+                    B921600 => BaudRate::B921600,
+                    b @ _ => unreachable!("Invalid baud constant: {}", b),
```

Comment:
I don't think this arm is necessary.  The compiler should be able to reason about this branch not being reachable and generate a compile error if something is off (which is better than this runtime error).